src/Affects/Models/Events/ConfigureModels.php

```diff
@@ -29,53 +26,30 @@ class ConfigureModels
      */
     public $event;
 
-    /**
-     * Tenant connection resolver.
-     *
-     * @var ConnectionResolverInterface
-     */
-    public static $resolver;
-
     public function __construct(Switched $event)
     {
         $this->event = $event;
     }
 
     /**
-     * Forces specific models to use the tenant connection.
+     * Runs static functions on models.
      *
-     * @param array $models
-     * @param bool  $reset  ; reset the connection back to the default if no tenant identified.
+     * @param array  $models
+     * @param string $function
+     * @param array  $arguments
      *
      * @return $this
      */
-    public function onTenant(array $models, bool $reset = true)
+    public function staticCallOnModels(array $models, string $function, array $arguments)
     {
-        $resolver = count($models) ? $this->getResolver($reset) : null;
-
         foreach ($models as $model) {
             if (!class_exists($model)) {
                 throw new InvalidArgumentException("$model does not exist");
             }
 
-            forward_static_call([$model, 'setConnectionResolver'], $resolver);
+            forward_static_call([$model, $function], ...$arguments);
```

Comment:
I don't like this "wildcard" ability at all inside an event like this with such a weird method name.. I'm extremely curious why this is necessary at all.

Reply:
I want it to be as flexible as possible if people want to use the event to run methods on models, they should be able to, why would we only allow them to change the connectionResolver?